src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/ProcessDependencyResult.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.DotNet.UpgradeAssistant.Analysis;
+using Microsoft.DotNet.UpgradeAssistant.Dependencies;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages
+{
+    public class ProcessDependencyResult : IProcessResult
+    {
+        public IList<Run> RunProcessResult(string toolName, Dictionary<string, IDependencyAnalysisState> analysisResults)
+        {
+            if (analysisResults is null)
+            {
+                throw new ArgumentNullException(nameof(analysisResults));
+            }
+
+            var runs = new List<Run>();
+            foreach (var analysis in analysisResults)
+            {
+                var results = ExtractResults(analysis.Value);
+
+                var run = new Run()
+                {
+                    Tool = new()
+                    {
+                        Driver = new()
+                        {
+                            Name = string.Join(" ", toolName, analysis.Key)
+                        }
+                    },
+                    Results = results
+                };
+
+                runs.Add(run);
+            }
+
+            return runs;
+        }
+
+        public static IList<Result> ExtractResults(IDependencyAnalysisState? analysisState)
+        {
+            var results = new List<Result>();
+
+            if (analysisState is null || !analysisState.AreChangesRecommended)
+            {
+                results.Add(new()
+                {
+                    Message = "No package updates needed".ToMessage()
+                });
+            }
+            else
+            {
+                GetResults("References to Delete", analysisState.References.Deletions);
```

Comment:
It would be nice if the reason *why* dependencies needed added or removed was included too. That's not available in dependency state, though, so it would be a larger work item to enable that. This is probably fine for the first iteration, but we should consider filing an issue to give more detail in this report.

For that to work, the dependency analyzers would need a way to report more information about the changes they're recommending. I think that will be useful, though. There could be a notification service that those types call to log messages like "Replace this version of package X with this other version of package X so that it supports the new TFM" or something like that and then this could get those messages from the notification service.